js/data/array_utils.js

```diff
@@ -108,7 +124,13 @@ function update(keyInfo, array, key, data, isBatch) {
             if(index < 0) {
                 return !isBatch && rejectedPromise(errors.Error('E4009'));
             }
+
             target = array[index];
+
+            if(immutable === true && isDefined(target) && isPlainObject(target)) {
```

Comment:
Why is isPlainObject? What is about class objects? Sometimes users use classes instead of plaint objects in DataGrid. 
I think we should use the following utils method: https://github.com/DevExpress/DevExtreme/blob/42ce857c16b4c64ca19cd7042b132ce8fb247eca/js/ui/grid_core/ui.grid_core.utils.js#L412.

Reply:
OK.